private/bufpkg/bufconnect/interceptors.go

```diff
@@ -42,38 +43,58 @@ func NewSetCLIVersionInterceptor(version string) connect.UnaryInterceptorFunc {
 	return interceptor
 }
 
+// MachineFinder finds the machine by name
+type MachineFinder interface {
+	getMachineForName(name string) (netrc.Machine, error)
+}
```

Comment:
`NewAuthorizationInterceptorProvider` is refactored to the suggested implementation, except the definition of `RemoteToken(address string) (token string, setFromEnvVar bool)` of `TokenFinder`. The `setFromEnvVar` is true when the remote token returned is obtained from the `tokenEnvKey` environment variable. This is needed since we need to return different error message [here](https://github.com/bufbuild/buf/blob/13e8ff16083aad6ef2f4e3543d0669857279ebc6/private/buf/bufcli/errors.go#L185) if the token is obtained from the environment variable.

Reply:
This makes the `NetrcTokens` a little bit hard to test. So I made it as an interface and added `netrcTokensImpl` implementation for testing.